src/ImageSharp.Drawing/Processors/DrawImageEffectProcessor.cs

```diff
@@ -0,0 +1,102 @@
+﻿// <copyright file="DrawImageEffectProcessor.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.Drawing.Processors
+{
+    using System;
+    using System.Numerics;
+    using System.Threading.Tasks;
+    using ImageSharp.PixelFormats;
+    using ImageSharp.Processing;
+
+    /// <summary>
+    /// Combines two images together by blending the pixels.
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal class DrawImageEffectProcessor<TPixel> : ImageProcessor<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DrawImageEffectProcessor{TPixel}"/> class.
+        /// </summary>
+        /// <param name="image">The image to blend with the currently processing image.</param>
+        /// <param name="size">The size to draw the blended image.</param>
+        /// <param name="location">The location to draw the blended image.</param>
+        /// <param name="pixelFunction">Pixel function effect to apply on every pixel</param>
+        /// <param name="alpha">The opacity of the image to blend. Between 0 and 100.</param>
+        public DrawImageEffectProcessor(Image<TPixel> image, Size size, Point location, Func<TPixel, TPixel, float, TPixel> pixelFunction, int alpha = 100)
+        {
+            Guard.MustBeBetweenOrEqualTo(alpha, 0, 100, nameof(alpha));
+            this.Image = image;
+            this.PixelFunction = pixelFunction;
+            this.Size = size;
+            this.Location = location;
+            this.Alpha = alpha;
+        }
+
+        /// <summary>
+        /// Gets the image to blend.
+        /// </summary>
+        public Image<TPixel> Image { get; private set; }
+
+        /// <summary>
+        /// Gets The function effect to apply on a per pixel basis
+        /// </summary>
+        public Func<TPixel, TPixel, float, TPixel> PixelFunction { get; private set; }

```

Comment:
With my proposal this would be `PixelTransform<TPixel, TPixel> PixelTransform { get; private set; }`